base_tier_validation/models/res_users.py

```diff
@@ -13,19 +13,22 @@ class Users(models.Model):
     def review_user_count(self):
         user_reviews = {}
         to_review_docs = {}
-        for review in self.env.user.review_ids.filtered(
-            lambda r: r.status == "pending"
-        ):
+        reviews = self.env["tier.review"].search(
+            [
+                ("id", "in", self.env.user.review_ids.ids),
+                ("status", "=", "pending"),
+                ("can_review", "=", True),
+            ]
+        )
+        for review in reviews:
```

Comment:
Why you use this code instead 
`for review in self.env.user.review_ids.filtered(lambda r: r.status == "pending" and r.can_review):` 
?

Reply:
Because I think the performance will be better then using filtered()

Reply:
I think the use of o2m fields has also some performance optimizations, have you compared both options?

Reply:
No, but I think when time passed, the user.review_ids can have so much records. Let's say 100,000 records. The filter operation can take up time (it will go through record one by one).
But if we use search with `status` and `can_review`, it can reduce a lot of records already.

My assumption is filter() is slower than searching a field. Is it? What is your opinion to this, I am open.

Reply:
Let's keep your change, intuitively it looks that it will be better performance-wise.